### PR TITLE
Add spec for auto-tupling of n-ary functions

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: docs/docs/reference/auto-parameter-tupling-spec.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+and you want to map `xs` to a list of `Int`s so that each pair of numbers is mapped to their sum. 
+Previously, the best way to do this was with a pattern-matching decomposition:
+```scala
+xs.map {
+  case (x, y) => x + y
+}
+```
+While correct, this is also inconvenient. Instead, automatic tupling of function parameters proposed to be able to write it the following way:
```
[the remark this answers is not in the document]
Comment (liufengyun) [reply docs/docs/reference/auto-parameter-tupling-spec.md]: > automatic tupling of function parameters proposed to be able ...
This does not read smoothly.

File: docs/docs/reference/auto-parameter-tupling-spec.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+or, equivalently:
+```scala
+xs.map(_ + _)
+```
+
+Generally, a function value with `n > 1` parameters can be converted to a pattern-matching closure using case if the expected type is a unary function type of the form `((T_1, ..., T_n)) => U`.
+
+### Type Checking
+
+Let a function `F` of the form `F = (p1, ..., pn) => E` for `n != 1`, parameters `p1, ..., pn`, and an expression `E`.
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply docs/docs/reference/auto-parameter-tupling-spec.md]: It was stated like that in the issue

[the remark this answers is not in the document]
Comment (nicolasstucki) [reply docs/docs/reference/auto-parameter-tupling-spec.md]: I will change it

**Comment by nicolasstucki** [comment]
@liufengyun could you have at the diff?

File: docs/sidebar.yml [hunk 3]
```diff
@@ -53,6 +53,8 @@ sidebar:
               url: docs/reference/implicit-by-name-parameters.html
             - title: Auto Parameter Tupling
               url: docs/reference/auto-parameter-tupling.html
+            - title: Auto Parameter Tupling - More Details
```
Comment (allanrenucci) [inline docs/sidebar.yml]: Following Martin's [comment](https://github.com/lampepfl/dotty/pull/5217#issuecomment-429864003), this should be removed

File: docs/docs/reference/auto-parameter-tupling-spec.md [hunk 4]
```diff
@@ -18,7 +18,7 @@ xs.map {
   case (x, y) => x + y
 }
 ```
-While correct, this is also inconvenient. Instead, automatic tupling of function parameters proposed to be able to write it the following way:
+While correct, this is inconvenient. Instead, we propose to be able to write it the following way:
```
Comment (liufengyun) [inline docs/docs/reference/auto-parameter-tupling-spec.md]: It seems `to be able` can be removed.

**Pull request merged**